File: app/src/main/java/org/tiny/record/view/splash/SplashActivity.java
package org.tiny.record.view.splash;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.alibaba.android.arouter.launcher.ARouter;

import org.fork.annotations.ForkPresenter;
import org.tiny.lib.core.BaseActivity;
import org.tiny.lib.core.Fork;
import org.tiny.record.R;
import org.tiny.record.databinding.ActivitySplashBinding;

/**
 * Created by tiny on 4/21/2018
 */
@ForkPresenter(SplashPresenter.class)
public class SplashActivity extends BaseActivity<SplashPresenter, ActivitySplashBinding> implements
        SplashContract.IView {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Fork.bind(this, R.layout.activity_splash);


        ARouter.getInstance().build("/test/activity").navigation();
    }
}

File: app/src/main/java/org/tiny/record/view/splash/SplashActivity.kt
package org.tiny.record.view.splash

import android.content.Intent
import android.os.Bundle
import android.view.WindowManager
import com.apkfuns.logutils.LogUtils
import com.baidu.location.Poi
import org.tiny.lib.core.BaseActivity
import org.tiny.record.R
import org.tiny.record.databinding.ActivitySplashBinding
import org.tiny.record.view.main.MainActivity

/**
 * Created by tiny on 2/10/2018
 */
class SplashActivity : BaseActivity<ActivitySplashBinding, SplashPresenter>(), SplashContract.IView {

    override fun createPresenter(): SplashPresenter? {
        return SplashPresenter(this)
    }

    override fun createLayoutId(): Int {
        return -1
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_LAYOUT_IN_SCREEN)
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS)

        mPresenter!!.limit()
        mPresenter!!.run()
    }

    override fun skip() {
        finishActivity()
    }

    override fun start(locations: ArrayList<Poi>) {
        locations.forEach({ point ->
            LogUtils.d("location: " + point.name)
        })
        finishActivity()
    }

    private fun finishActivity() {
        startActivity(Intent(this, MainActivity::class.java))
        finish()
        overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
    }
}